### PR TITLE
Update the mock Api usage configuration attribute

### PR DESCRIPTION
**Pull request by SaurabhMSFT**
The configuration attribute for using mock Api in native auth has been changed to useMockApiForNativeAuth from useRealAuthority

### CODE REVIEW

File: msal/src/androidTest/java/com/microsoft/identity/nativeauth/NativeAuthPublicClientApplicationConfigurationFactoryTest.kt [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    }
+
+    @Test
+    fun testNativeAuthorityConfigMockApiFalse() {
+        runBlocking {
+            try {
+                var config = NativeAuthPublicClientApplicationConfigurationFactory.initializeNativeAuthConfiguration(
+                    context, R.raw.test_native_auth_use_mock_api_false)
+
+                Assert.assertEquals(false, config.useMockAuthority)
```
Comment (SammyO) [inline msal/src/androidTest/java/com/microsoft/identity/nativeauth/NativeAuthPublicClientApplicationConfigurationFactoryTest.kt:98]:
```suggestion
                Assert.assertFalse(config.useMockAuthority)
```

File: msal/src/androidTest/java/com/microsoft/identity/nativeauth/NativeAuthPublicClientApplicationConfigurationFactoryTest.kt [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+                Assert.assertEquals(false, BuildValues.shouldUseMockApiForNativeAuth())
```
Comment (SammyO) [inline msal/src/androidTest/java/com/microsoft/identity/nativeauth/NativeAuthPublicClientApplicationConfigurationFactoryTest.kt]:
```suggestion
                Assert.assertFalse(BuildValues.shouldUseMockApiForNativeAuth())
```

File: msal/src/androidTest/java/com/microsoft/identity/nativeauth/NativeAuthPublicClientApplicationConfigurationFactoryTest.kt [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    }
+
+    @Test
+    fun testNativeAuthorityConfigMockApiTrue() {
+        runBlocking {
+            try {
+                var config = NativeAuthPublicClientApplicationConfigurationFactory.initializeNativeAuthConfiguration(
+                    context, R.raw.test_native_auth_use_mock_api_true)
+
+                Assert.assertEquals(true, config.useMockAuthority)
```
Comment (SammyO) [inline msal/src/androidTest/java/com/microsoft/identity/nativeauth/NativeAuthPublicClientApplicationConfigurationFactoryTest.kt:83]:
```suggestion
                Assert.assertTrue(config.useMockAuthority)
```

File: msal/src/androidTest/java/com/microsoft/identity/nativeauth/NativeAuthPublicClientApplicationConfigurationFactoryTest.kt [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+                Assert.assertEquals(true, BuildValues.shouldUseMockApiForNativeAuth())
```
Comment (SammyO) [inline msal/src/androidTest/java/com/microsoft/identity/nativeauth/NativeAuthPublicClientApplicationConfigurationFactoryTest.kt]:
```suggestion
                Assert.assertTrue(BuildValues.shouldUseMockApiForNativeAuth())
```

File: msal/src/androidTest/java/com/microsoft/identity/nativeauth/NativeAuthPublicClientApplicationConfigurationFactoryTest.kt [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+                .path
+        )
+
+        context = InstrumentationRegistry.getInstrumentation().context.applicationContext
+        cachedUseMockApi = BuildValues.shouldUseMockApiForNativeAuth()
+    }
+
+    @After
+    fun tearDown() {
+        BuildValues.setUseMockApiForNativeAuth(cachedUseMockApi)
```
Comment (SammyO) [inline msal/src/androidTest/java/com/microsoft/identity/nativeauth/NativeAuthPublicClientApplicationConfigurationFactoryTest.kt]: Why is this necessary?

Comment (SaurabhMSFT) [reply msal/src/androidTest/java/com/microsoft/identity/nativeauth/NativeAuthPublicClientApplicationConfigurationFactoryTest.kt]: It isn't. I have removed it.

File: msal/src/androidTest/java/com/microsoft/identity/nativeauth/NativeAuthPublicClientApplicationConfigurationFactoryTest.kt (hunk 3 above)
Comment (SaurabhMSFT) [reply msal/src/androidTest/java/com/microsoft/identity/nativeauth/NativeAuthPublicClientApplicationConfigurationFactoryTest.kt:83]: useMockAuthority is nullable so I have kept assertEquals

File: msal/src/androidTest/java/com/microsoft/identity/nativeauth/NativeAuthPublicClientApplicationConfigurationFactoryTest.kt (hunk 4 above)
Comment (SaurabhMSFT) [reply msal/src/androidTest/java/com/microsoft/identity/nativeauth/NativeAuthPublicClientApplicationConfigurationFactoryTest.kt]: Done

File: msal/src/androidTest/java/com/microsoft/identity/nativeauth/NativeAuthPublicClientApplicationConfigurationFactoryTest.kt (hunk 1 above)
Comment (SaurabhMSFT) [reply msal/src/androidTest/java/com/microsoft/identity/nativeauth/NativeAuthPublicClientApplicationConfigurationFactoryTest.kt:98]: useMockAuthority is nullable so I have kept assertEquals

File: msal/src/androidTest/java/com/microsoft/identity/nativeauth/NativeAuthPublicClientApplicationConfigurationFactoryTest.kt (hunk 2 above)
Comment (SaurabhMSFT) [reply msal/src/androidTest/java/com/microsoft/identity/nativeauth/NativeAuthPublicClientApplicationConfigurationFactoryTest.kt]: done

**Review by SammyO — APPROVED** [review]

**Review by fadidurah — APPROVED** [review]

**Review by shahzaibj — APPROVED** [review]
